opal/core/episodes.py

```diff
@@ -63,20 +65,58 @@ def has_stage(self, stage):
         """
         return stage in self.get_stages()
 
-    def set_stage(self, stage, user, data):
+    @transaction.atomic
```

Comment:
Can we explain what the desired roll back behaviour / reason to explicitly add a sub-transaction here is?

Reply:
I guess I'm wondering if we shouldn't use it as a context manager to wrap those lines to make it more explicit that you want to make sure they're done together....